sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/decimalFunctions.scala

```diff
@@ -28,6 +29,15 @@ case class UnscaledValue(child: Expression) extends UnaryExpression {
   override def nullable: Boolean = child.nullable
   override def toString: String = s"UnscaledValue($child)"
 
+  override def checkInputDataTypes(): TypeCheckResult = {
+    if (child.dataType.isInstanceOf[DecimalType]) {
+      TypeCheckResult.TypeCheckSuccess
+    } else {
+      TypeCheckResult.TypeCheckFailure(
+        s"UnscaledValue only accept decimal type, not ${child.dataType}")
```

Comment:
`ExpectedInputTypes` not only define the type check behavior, but also type cast. Do we need to do type cast for `UnscaledValue` and `MakeDecimal`?